packages/runner-shared/src/event-manager.js

```diff
@@ -431,11 +431,15 @@ export const eventManager = {
     })
 
     Cypress.on('log:changed', (log) => {
-      const displayProps = Cypress.runner.getDisplayPropsForLog(log)
+      // Cypress logs will only trigger an update every 4 seconds so there is a
+      // chance the runner has been torn down when the update is triggered.
+      if (Cypress.runner) {
+        const displayProps = Cypress.runner.getDisplayPropsForLog(log)
 
-      this._interceptStudio(displayProps)
+        this._interceptStudio(displayProps)
 
-      reporterBus.emit('reporter:log:state:changed', displayProps)
+        reporterBus.emit('reporter:log:state:changed', displayProps)
+      }
```

Comment:
Could do `if (!Cypress.runner) return` here and get rid of the nesting.

Reply:
[`b87099c` (#20857)](https://github.com/cypress-io/cypress/pull/20857/commits/b87099ce916e6c2cceea0ed85b87df0758f2115e)